packages/core/src/error.ts

```diff
@@ -70,8 +70,9 @@ class Neo4jError extends Error {
    * @param {string} message - the error message
    * @param {string} code - Optional error code. Will be populated when error originates in the database.
    */
-  constructor (message: string, code: Neo4jErrorCode) {
-    super(message)
+  constructor (message: string, code: Neo4jErrorCode, cause?: Error) {
+    // @ts-expect-error
+    super(message, cause != null ? { cause } : undefined)
```

Comment:
I am surprised this doesn't require a poly-fill. Why is this, do you have a link to docs or similar where I can educate myself?

Reply:
The Error cause was added as an extra parameter to the Error constructor and Javascript doesn't enforce the number of parameters for calling a function. So, if i declare `function fn(a) { ... }` and you call as `fn('a', 'b')` without any issue. So, calling `Error('message', { cause: 'a' })` will not break any old implementation. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error

Reply:
JS for you :upside_down_face: Thank's for the explanation.